src/components/application_manager/src/commands/mobile/register_app_interface_request.cc

```diff
@@ -501,6 +509,242 @@ void FillUIRelatedFields(smart_objects::SmartObject& response_params,
       hmi_capabilities.rc_supported();
 }
 
+void RegisterAppInterfaceRequest::SendRegisterAppInterfaceResponseToMobile(
+    ApplicationType app_type) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  smart_objects::SmartObject response_params(smart_objects::SmartType_Map);
+
+  mobile_apis::Result::eType result_code = mobile_apis::Result::SUCCESS;
+
+  const HMICapabilities& hmi_capabilities =
+      application_manager_.hmi_capabilities();
+
+  const uint32_t key = connection_key();
+  ApplicationSharedPtr application = application_manager_.application(key);
+
+  resumption::ResumeCtrl& resumer = application_manager_.resume_controller();
+
+  if (!application) {
+    LOG4CXX_ERROR(logger_,
+                  "There is no application for such connection key" << key);
+    LOG4CXX_DEBUG(logger_, "Need to start resume data persistent timer");
+    resumer.OnAppRegistrationEnd();
+    return;
+  }
+
+  response_params[strings::sync_msg_version][strings::major_version] =
+      major_version;  // From generated file interfaces/generated_msg_version.h
+  response_params[strings::sync_msg_version][strings::minor_version] =
+      minor_version;  // From generated file interfaces/generated_msg_version.h
+  response_params[strings::sync_msg_version][strings::patch_version] =
+      patch_version;  // From generated file interfaces/generated_msg_version.h
+
+  const smart_objects::SmartObject& msg_params =
+      (*message_)[strings::msg_params];
+
+  if (msg_params[strings::language_desired].asInt() !=
+          hmi_capabilities.active_vr_language() ||
```

Comment:
@yang1070 will investigate what is VR is not supported by HU.
In case if there are going to some bug, it should be fixed after delivery. 
Priority of such issue should be medium because it is not trivial use case if HMI has no VR support.

Reply:
@yang1070 @LuxoftAKutsan 
Basically this method mostly duplicates logic from deprecated RegisterAppInterfaceRequest::SendRegisterAppInterfaceResponseToMobile() including this line. So even if there is some issue it should be reproducible with prior releases as well and should not be considered as regression introduced by PR.

Reply:
do not agree. copy-paste errors are still errors need to be fixed. 
It is a new function, new code. shall resolve the issue.

Reply:
This is not a completely copy-pase issue. Method SendRegisterAppInterfaceResponseToMobile was changed. And the signature of the method was changed. But due to our deprecation policies we are unable just to change function signature, Function should be copied, and old version should be marked as DEPRECATED. Anyway, this is a bug and should be fixed. Luxoft will create issue to fix that bug, and we will reply with issue number to this thread.

Reply:
@yang1070 I investigated this case and can see that there is no issue.
Even if there is no VR support on HU, SDL will by default setup EN-US language as active VR:
```
if (check_existing_json_member(root_json, "VR")) {
      Json::Value vr = root_json.get("VR", "");
          if (check_existing_json_member(vr, "language")) {
            const std::string lang = vr.get("language", "EN-US").asString();
            set_active_vr_language(MessageHelper::CommonLanguageFromString(lang));
          } else {
            set_active_vr_language(
                MessageHelper::CommonLanguageFromString("EN-US"));
          }
```
 language, and response to HMI with success result code on registration, with hmi_capabilities.
We create the test script to check the sequence. It is available in  https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/1757/

Reply:
Issue for investigating has been created: https://github.com/smartdevicelink/sdl_core/issues/1944